resolwe/flow/models/entity.py

```diff
@@ -46,39 +46,110 @@ def __str__(self):
 
 
 class Relation(BaseModel):
-    """Relations between entities."""
+    """Relations between entities.
+
+    The Relation model defines the associations and dependencies between
+    entities in a given collection:
+
+    .. code-block:: json
+
+        {
+            "collection": "<collection_id>",
+            "type": "comparison",
+            "category": "case-control study",
+            "entities": [
+                {"enetity": "<entity1_id>", "label": "control"},
+                {"enetity": "<entity2_id>", "label": "case"},
+                {"enetity": "<entity3_id>", "label": "case"}
+            ]
+        }
+
+    Relation ``type`` defines a specific set of associations among
+    entities. It can be something like ``group``, ``comparison`` or
+    ``series``. The relation ``type`` is an instance of
+    :class:`.RelationType` and should be defined in any Django app that
+    uses relations (e.g., as a fixture). Multiple relations of the same
+    type are allowed on the collection.
+
+    Relation ``category`` defines a specific use case. The relation
+    category must be unique in a collection, so that users can
+    distinguish between different relations. In the example above, we
+    could add another ``comparison`` relation of ``category``, say
+    ``Case-case study`` to compare ``<entity1>`` with ``<entity2>``.
+
+    Relation is linked to :class:`resolwe.flow.models.Collection` to
+    enable defining different relations structures in different
+    collections. This also greatly speed up retrieving of relations, as
+    they are envisioned to be mainly used on a collection level.
+
+    ``unit`` defines units used in partitions where it is applicable,
+    e.g. in relations of type ``series``.
+
+    """
+
+    UNIT_SECOND = 's'
+    UNIT_MINUTE = 'min'
+    UNIT_HOUR = 'hr'
+    UNIT_DAY = 'd'
+    UNIT_WEEK = 'wk'
+    UNIT_CHOICES = (
+        (UNIT_SECOND, "Second"),
+        (UNIT_MINUTE, "Minute"),
+        (UNIT_HOUR, "Hour"),
+        (UNIT_DAY, "Day"),
+        (UNIT_WEEK, "Week"),
+    )
 
-    class Meta(BaseCollection.Meta):
+    #: type of the relation
+    type = models.ForeignKey('RelationType', on_delete=models.PROTECT)
+
+    #: collection to which relation belongs
+    collection = models.ForeignKey('Collection', on_delete=models.CASCADE)
+
+    #: partitions of entities in the relation
+    entities = models.ManyToManyField(Entity, through='RelationPartition')
+
+    #: category of the relation
+    category = models.CharField(max_length=100)
+
+    #: unit used in the partitions' positions (where applicable, e.g. for serieses)
+    unit = models.CharField(max_length=3, choices=UNIT_CHOICES, null=True, blank=True)
+
+    def save(self, *args, **kwargs):
+        """Lowercase the ``category`` field and save the model."""
+        self.category = self.category.lower()
```

Comment:
Is it ok to save category as lowercase? You loose some information.

Reply:
Maybe the category should be saved as it is and fix the filter to be case insensitive.

Reply:
This was fixed in #532